build-logic/basics/src/main/kotlin/gradlebuild/basics/BuildParams.kt

```diff
@@ -0,0 +1,274 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gradlebuild.basics
+
+import gradlebuild.basics.BuildParams.BUILD_BRANCH
+import gradlebuild.basics.BuildParams.BUILD_COMMIT_ID
+import gradlebuild.basics.BuildParams.BUILD_CONFIGURATION_ID
+import gradlebuild.basics.BuildParams.BUILD_FINAL_RELEASE
+import gradlebuild.basics.BuildParams.BUILD_ID
+import gradlebuild.basics.BuildParams.BUILD_IGNORE_INCOMING_BUILD_RECEIPT
+import gradlebuild.basics.BuildParams.BUILD_MILESTONE_NUMBER
+import gradlebuild.basics.BuildParams.BUILD_PROMOTION_COMMIT_ID
+import gradlebuild.basics.BuildParams.BUILD_RC_NUMBER
+import gradlebuild.basics.BuildParams.BUILD_SERVER_URL
+import gradlebuild.basics.BuildParams.BUILD_TIMESTAMP
+import gradlebuild.basics.BuildParams.BUILD_VCS_NUMBER
+import gradlebuild.basics.BuildParams.BUILD_VERSION_QUALIFIER
+import gradlebuild.basics.BuildParams.CI_ENVIRONMENT_VARIABLE
+import gradlebuild.basics.BuildParams.GRADLE_INSTALL_PATH
+import gradlebuild.basics.BuildParams.INCLUDE_PERFORMANCE_TEST_SCENARIOS
+import gradlebuild.basics.BuildParams.MAX_PARALLEL_FORKS
+import gradlebuild.basics.BuildParams.PERFORMANCE_BASELINES
+import gradlebuild.basics.BuildParams.PERFORMANCE_DB_PASSWORD
+import gradlebuild.basics.BuildParams.PERFORMANCE_DB_PASSWORD_ENV
+import gradlebuild.basics.BuildParams.PERFORMANCE_DB_URL
+import gradlebuild.basics.BuildParams.PERFORMANCE_DB_USERNAME
+import gradlebuild.basics.BuildParams.PERFORMANCE_DEPENDENCY_BUILD_IDS
+import gradlebuild.basics.BuildParams.PERFORMANCE_MAX_PROJECTS
+import gradlebuild.basics.BuildParams.PERFORMANCE_TEST_VERBOSE
+import gradlebuild.basics.BuildParams.RERUN_ALL_TESTS
+import gradlebuild.basics.BuildParams.TEST_DISTRIBUTION_ENABLED
+import gradlebuild.basics.BuildParams.TEST_DISTRIBUTION_PARTITION_SIZE
+import gradlebuild.basics.BuildParams.TEST_FILTERING_ENABLED
+import gradlebuild.basics.BuildParams.TEST_JAVA_VENDOR
+import gradlebuild.basics.BuildParams.TEST_JAVA_VERSION
+import gradlebuild.basics.BuildParams.TEST_SPLIT_EXCLUDE_TEST_CLASSES
+import gradlebuild.basics.BuildParams.TEST_SPLIT_INCLUDE_TEST_CLASSES
+import gradlebuild.basics.BuildParams.TEST_SPLIT_ONLY_TEST_GRADLE_VERSION
+import org.gradle.api.JavaVersion
+import org.gradle.api.Project
+import org.gradle.api.provider.Provider
+
+
+object BuildParams {
+    const val BUILD_BRANCH = "BUILD_BRANCH"
```

Comment:
We probably shouldn't put those CI environment variables in the same place we put the feature switches for our build. Though I am not sure, yet.

What we should have is documentation for all the constants.